### ISSUE TITLE
Unsynchronized sign-in doesn't save

### ISSUE THREAD

**Issue by theredwillow** [jsfiddle/jsfiddle-issues#1465]
I entered a new JSFiddle without having signed in first and created some work. You have to be logged in to save work, but I was worried I'd lose my work by clicking sign-in so I opened it in a new tab and registered. When I returned to the tab with my working JSFiddle, the sign-in state was **not synchronized**. Unable to save my work, I resorted to clicking the sign-in button anyways and it refreshed the new JSFiddle page and I lost all my work for the afternoon.

**Comment by oskarkrawczyk** [comment]
1. You are able to save when not logged in
2. After logging in a different tab, (as above) you can save, and it'll be saved under your account
3. You can bring back unsaved changes via the **Unsaved draft** action (in red so people don't miss it)

There are no plans to automatically asynchronously change the login state of all opened JSFiddle instances. Closing.

**Comment by q00u** [comment]
I entered a new JSFiddle without having signed in first and created some work. I went to save, and it said that if I wanted to delete this in the future, I should sign in.

Worried that I'd lose my work by clicking sign-in, so I opend a new tab and signed in there.

When I returned to the tab with my working JSFiddle, the sign-in state was not synched.

Clicking "run" had appeared to present a new state in history, so I _assumed_ refreshing was a safe action. I refreshed the page, to synch my sign-in state.

I was presented a new, empty jsfiddle. My work was gone.

This is a usability nightmare.

Can I go back a page and see my work in the page history? No, go back and it's now a fresh empty page.

**Unsaved draft** is where?

JSFiddle shouldn't throw open work away if you refresh the page.
